terraso_backend/apps/graphql/schema/schema.graphql

```diff
@@ -1527,6 +1528,106 @@ String, Boolean, Int, Float, List or Object.
 """
 scalar GenericScalar
 
+"""Soil ID algorithm queries."""
+type SoilID {
+  locationBasedSoilMatches(latitude: Float!, longitude: Float!): LocationBasedSoilMatches!
+  dataBasedSoilMatches(latitude: Float!, longitude: Float!, data: SoilIDInputData!): DataBasedSoilMatches!
+}
+
+"""A ranked group of soil matches based solely on a coordinate pair."""
+type LocationBasedSoilMatches {
+  matches: [LocationBasedSoilMatch]!
+}
+
+"""A soil match based solely on a coordinate pair."""
+type LocationBasedSoilMatch {
+  soilInfo: SoilInfo!
+  match: SoilMatch!
+}
+
+"""Provides information about soil at a particular location."""
+type SoilInfo {
+  dataSource: String!
+  soilSeries: SoilSeries!
+  ecologicalSite: EcologicalSite
+  landCapabilityClass: LandCapabilityClass!
+  soilData: [SoilIdSoilData!]!
+}
+
+"""Information about a soil series."""
+type SoilSeries {
+  name: String!
+  taxonomySubgroup: String!
+  description: String!
+  fullDescriptionUrl: String!
+}
+
+"""Information about an ecological site."""
+type EcologicalSite {
+  name: String!
+  id: String!
+  url: String!
+}
+
+"""Caveat: may want to update these fields to an enum at some point."""
+type LandCapabilityClass {
+  capabilityClass: String!
+  subClass: String!
+}
+
+type SoilIdSoilData {
+  depthInterval: DepthInterval!
+  texture: SoilIdDepthDependentSoilDataTextureChoices
+  rockFragmentVolume: SoilIdDepthDependentSoilDataRockFragmentVolumeChoices
+  colorHue: Float
+  colorValue: Float
+  colorChroma: Float
+}
+
+"""
+The likelihood score and rank within the match group for a particular soil type.
+"""
+type SoilMatch {
+  score: Float!
```

Comment:
I think we need another piece of data here indicating whether the match is coming from inside the soil map unit or some adjacent unit. (This is specified in the design for the match info sheet).

Reply:
ooh good call!